src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -442,8 +418,8 @@ private unsafe void RejectRequestLine(byte* requestLine, int length)
         }
 
         [StackTraceHidden]
-        private unsafe void RejectRequestHeader(byte* headerLine, int length)
-            => throw GetInvalidRequestException(RequestRejectionReason.InvalidRequestHeader, headerLine, length);
+        private void RejectRequestHeader(ReadOnlySpan<byte> headerLine)
+            => throw GetInvalidRequestException(RequestRejectionReason.InvalidRequestHeader, headerLine);
 
         [StackTraceHidden]
         private unsafe void RejectUnknownVersion(byte* version, int length)
```

Comment:
Make safe?

Reply:
Is used by start line parsing rather than header parsing; so would be out of scope for this PR.

e.g. moved this change out of the startline changes as it was getting a bit big